### PR TITLE
Modularise sonobi

### PR DESCRIPTION
**Pull request by rich-nguyen**
I've separated `commercial/modules/dfp/init` into:

- `commercial/modules/dfp/prepare-sonobi-tag`
- `commercial/modules/dfp/prepare-googletag`

because it allows us to clearly measure the impact of one module on the other, now that sonobi is 100%. The sonobi has to finish loading before the googletag module can start. This PR makes sure our commercial timing system reflects the offset of googletag's execution caused by sonobi.

I've removed the ophan commercial advert tracking, it is not used anymore, can't be viewed, and will save us some aws running cost, especially for @obrienm 's team.

I've also renamed `commercial/modules/dfp/load` to `commercial/modules/dfp/fill-advert-slots`, for more descriptive analytics reporting.

### CODE REVIEW

File: static/src/javascripts/projects/commercial/modules/dfp/performance-logging.js [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+        var timerStart = userTiming.getCurrentTime();
+        performanceLog.modules.push({
+            name: moduleName,
+            start: timerStart
+        });
+    }
+
+    function moduleEnd(moduleName) {
+        var timerEnd = userTiming.getCurrentTime();
+        var module = performanceLog.modules.find(function(module){
```
Comment (regiskuckaertz) [inline static/src/javascripts/projects/commercial/modules/dfp/performance-logging.js]: `Array.prototype.find` needs a shim in older browsers. Do we provide it?

File: static/src/javascripts/projects/commercial/modules/dfp/ophan-tracking.js [hunk 2]
```diff
@@ -1,205 +0,0 @@
...
-                    4: 'receive',
-                    6: 'render'
-                },
-                lifecycleIdToAdvertTiming = {
-                    3: 'dfpFetching',
-                    4: 'dfpReceived',
-                    6: 'dfpRendered'
-                };
-
-            googletag.debug_log.log = function interceptedGptDebugger(level, message, service, slot) {
```
Comment (regiskuckaertz) [inline static/src/javascripts/projects/commercial/modules/dfp/ophan-tracking.js]: Good riddance 🚤

File: static/src/javascripts/projects/commercial/modules/dfp/prepare-googletag.js [hunk 3]
```diff
@@ -33,64 +34,70 @@ define([
...
+                    }
+                }
 
-            if (dfpEnv.sonobiEnabled) {
-                // Just load googletag. Sonobi's wrapper will already be loaded, and googletag is already added to the window by sonobi.
-                require(['js!googletag.js']);
-                ophanTracking.addTag('sonobi');
-            } else {
-                require(['js!googletag.js']);
+                window.googletag.cmd.push = raven.wrap({deep: true}, window.googletag.cmd.push);
```
Comment (regiskuckaertz) [inline static/src/javascripts/projects/commercial/modules/dfp/prepare-googletag.js]: I've always wondered whether this line made any sense at all. When `googletag.js` loads, it overwrites `googletag.cmd.push`. Anyway, that's totally unrelated 😆

File: static/src/javascripts/projects/commercial/modules/dfp/prepare-googletag.js [hunk 4]
```diff
@@ -33,64 +34,70 @@ define([
...
+                    setPageTargeting,
+                    moduleCompleted,
+                    resolve
+                );
+            });
+        }
 
-            window.googletag.cmd.push = raven.wrap({deep: true}, window.googletag.cmd.push);
+        if (commercialFeatures.dfpAdvertising) {
+            return prepareSonobiTag.init().then(setupAdvertising).catch(function(){
```
Comment (regiskuckaertz) [inline static/src/javascripts/projects/commercial/modules/dfp/prepare-googletag.js]: Hasn't `prepareSonobiTag` already been called at this point? (in `commercial.js`)

File: static/src/javascripts/projects/commercial/modules/dfp/prepare-sonobi-tag.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    'common/modules/commercial/commercial-features',
+    'commercial/modules/dfp/dfp-env'
+], function(
+    commercialFeatures,
+    dfpEnv
+){
+    var prom = null;
+
+    function setupSonobi() {
+        return new Promise.resolve(require(['js!sonobi.js']));
```
Comment (regiskuckaertz) [inline static/src/javascripts/projects/commercial/modules/dfp/prepare-sonobi-tag.js]: You don't need the `new` here

Comment (regiskuckaertz) [inline static/src/javascripts/projects/commercial/modules/dfp/prepare-sonobi-tag.js]: (also don't forget to import the `Promise` module)

File: static/src/javascripts/bootstraps/commercial.js [hunk 6]
```diff
@@ -97,17 +100,21 @@ define([
...
         modules.forEach(function (pair) {
 
             var moduleName = pair[0];
+            var moduleInit = pair[1];
+            var hasCustomTiming = pair[2];
 
             robust.catchErrorsAndLog(moduleName, function () {
-                var modulePromise = pair[1]().then(function(){
-                    ophanTracking.moduleCheckpoint(moduleName, baseline);
+                var modulePromise = moduleInit(moduleName).then(function(){
```
Comment (rich-nguyen) [inline static/src/javascripts/bootstraps/commercial.js]: The only thing I could think of was moving the name each module into the module itself, but it doesn't really help, and it might even be harder to see.

File: static/src/javascripts/projects/commercial/modules/dfp/prepare-googletag.js (hunk 3 above)
Comment (rich-nguyen) [inline static/src/javascripts/projects/commercial/modules/dfp/prepare-googletag.js]: yeah, good point, this probably doesn't do what was intended. could be another pr to think of a better way to protect our system from the cmd buffer.

File: static/src/javascripts/projects/commercial/modules/dfp/performance-logging.js [hunk 7]
```diff
@@ -49,10 +49,13 @@ define([
...
     function moduleEnd(moduleName) {
         var timerEnd = userTiming.getCurrentTime();
-        var module = performanceLog.modules.find(function(module){
-            return module.name === moduleName;
-        });
-        if (module) {
+
+        var moduleIndex = performanceLog.modules.map(function (module) {
+            return module.name;
+        }).indexOf(moduleName);
```
Comment (regiskuckaertz) [inline static/src/javascripts/projects/commercial/modules/dfp/performance-logging.js]: That's why @janua polyfill service is so awesome, you can (almost) forget about browser compat.

File: static/src/javascripts/projects/commercial/modules/dfp/prepare-googletag.js (hunk 4 above)
Comment (rich-nguyen) [inline static/src/javascripts/projects/commercial/modules/dfp/prepare-googletag.js]: Yes it is. I thought about this a bit, and decided it best to execute both the sonobi module and the googletag module in the commercial bootstrap, but maintain the dependency between the two, within the depending module. This means that the sonobi module is tracked and measured in the same way as other primary modules, whilst I can still individually measure this googletag module.

File: static/src/javascripts/projects/commercial/modules/dfp/prepare-sonobi-tag.js (hunk 5 above)
Comment (rich-nguyen) [inline static/src/javascripts/projects/commercial/modules/dfp/prepare-sonobi-tag.js]: good catch, thanks

**Review by regiskuckaertz — APPROVED** [review]
Great!

File: static/src/javascripts/projects/commercial/modules/dfp/prepare-sonobi-tag.js [hunk 8]
```diff
@@ -0,0 +1,24 @@
+define([
+    'common/modules/commercial/commercial-features',
+    'commercial/modules/dfp/dfp-env'
+], function(
+    commercialFeatures,
+    dfpEnv
+){
+    var prom = null;
```
Comment (rich-nguyen) [inline static/src/javascripts/projects/commercial/modules/dfp/prepare-sonobi-tag.js]: Cool, very tidy. I guess `require` is memoized, in a sense. what i found was i could memoize `setupSonobi`, but if i memoized `init`, the second call still executed. do you know why?
